src/main.rs

```diff
@@ -212,6 +221,14 @@ fn entry() -> Result<(), String> {
                     println!("{tag}");
                 }
             }
+            let error_flag_passed = matches
+                .subcommand
+                .unwrap() // Safe because we're _in_ a subcommand
+                .matches
+                .is_present(LIST_UNUSED_ERROR_OPTION);
+            if error_flag_passed && !tags_map.lock().unwrap().is_empty() {
+                std::process::exit(1);
```

Comment:
The `entry()` function already has a way to return errors (the `Result` return type). Lets use that instead of killing the process to ensure any cleanup is done properly.